### PR TITLE
Fix UnboundLocalError in invert_attention_mask by adding proper shape…

### PR DESCRIPTION
**Pull request by gagandhakrey**
… validation

# What does this PR do?

Problem The invert_attention_mask function in src/transformers/modeling_utils.py crashed with an UnboundLocalError when given an encoder_attention_mask shape that wasn't exactly 2D or 3D. Because it only checked for 2D and 3D shapes using separate if statements without any elif or else fallback, the encoder_extended_attention_mask variable would remain unassigned before attempting a .to() cast, causing models with 1D or 4D inputs to fail silently with confusing stack traces.

Fix Refactored the second conditional check to use an elif and added a final else branch. If the passed tensor is neither 2D nor 3D, it now explicitly raises a ValueError detailing the incorrect tensor shape, matching the behavior of the nearby get_extended_attention_mask function and preventing the UnboundLocalError entirely.

Fixes # (issue)

## Code Agent Policy

The Transformers repo is currently being overwhelmed by a large number of PRs and issue comments written by
code agents. We are currently bottlenecked by our ability to review and respond to them. As a result, 
**we ask that new users do not submit pure code agent PRs** at this time. 
You may use code agents in drafting or to help you diagnose issues. We'd also ask autonomous "OpenClaw"-like agents
not to open any PRs or issues for the moment.

PRs that appear to be fully agent-written will probably be closed without review, and we may block users who do this
repeatedly or maliciously. 

This is a rapidly-evolving situation that's causing significant shockwaves in the open-source community. As a result, 
this policy is likely to be updated regularly in the near future. For more information, please read [`CONTRIBUTING.md`](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md).

- [ ] I confirm that this is not a pure code agent PR.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by zucchini-nlp — COMMENTED** [review]
Makes sense to me, though I think we were getting rid of attn mask fn here in favor of `masking_utils` @vasqu ?

File: src/transformers/modeling_utils.py [hunk 1]
```diff
@@ -867,8 +867,10 @@ def invert_attention_mask(self, encoder_attention_mask: Tensor) -> Tensor:
         """
         if encoder_attention_mask.dim() == 3:
             encoder_extended_attention_mask = encoder_attention_mask[:, None, :, :]
-        if encoder_attention_mask.dim() == 2:
+        elif encoder_attention_mask.dim() == 2:
             encoder_extended_attention_mask = encoder_attention_mask[:, None, None, :]
+        else:
+            raise ValueError(f"Wrong shape for encoder_attention_mask (shape {encoder_attention_mask.shape})")
```
Comment (vasqu) [inline src/transformers/modeling_utils.py:873]: Yea. we try to remove these tbh - see #43924 (I really should get back to that PR). So I honestly don't want to add extra logic that may result in other issues

**Comment by zucchini-nlp** [comment]
@gagandhakrey thanks a lot for your contrib. Since I don't see an actual model affected, ig we don't have real issues with models (luckily). We'll remove `invert_attn_mask` soon as mentioned, so let's close this in favor of https://github.com/huggingface/transformers/pull/43924
